easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,189 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for MXNet, implemented as an easyblock
+
+@author: Ward Poelmans (Free University of Brussels)
+"""
+import os
+import shutil
+from distutils.version import LooseVersion
+
+from easybuild.easyblocks.generic.makecp import MakeCp
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.easyblocks.generic.rpackage import RPackage
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import apply_regex_substitutions, copy_file, symlink, write_file, mkdir
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_MXNet(MakeCp):
+    """Easyblock to build and install MXNet"""
+
+    @staticmethod
+    def extra_options():
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def __init__(self, *args, **kwargs):
+        """Initialize custom class variables."""
+        super(EB_MXNet, self).__init__(*args, **kwargs)
+
+        self.mxnet_src_dir = None
+        self.py_ext = PythonPackage(self, {'name': 'mxnet'})
+        self.py_ext.module_generator = self.module_generator
+        self.r_ext = RPackage(self, {'name': 'mxnet'})
+        self.r_ext.module_generator = self.module_generator
+
+    def extract_step(self):
+        """
+        Prepare a combined MXNet source tree. Move all submodules
+        to their right place.
+        """
+        # Extract everything into separate directories.
+        super(EB_MXNet, self).extract_step()
+
+        # Find the full path to the directory that was unpacked from mxnet-*.tar.gz.
+        for srcdir in os.listdir(self.builddir):
+            if srcdir.startswith("mxnet-"):
+                self.mxnet_src_dir = os.path.join(self.builddir, srcdir)
+                break
+
+        if self.mxnet_src_dir is None:
+            raise EasyBuildError("Could not find the MXNet source directory")
+
+        self.log.debug("MXNet dir is: %s", self.mxnet_src_dir)
+
+        for srcdir in os.listdir(self.builddir):
+            if srcdir.startswith("mxnet-"):
+                continue
+            else:
+                submodule, _, _ = srcdir.rpartition('-')
+                try:
+                    newdir = os.path.join(self.mxnet_src_dir, submodule)
+                    olddir = os.path.join(self.builddir, srcdir)
+                    # first remove empty existing directory
+                    os.rmdir(newdir)
```

Comment:
use http://easybuild.readthedocs.io/en/latest/api/easybuild.tools.filetools.html#easybuild.tools.filetools.rmtree2 (outside of `try`)